packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -233,12 +246,35 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
     debug(`${deployName}: deploying...`);
   }
 
-  const executionId = uuid.v4();
   const bodyParameter = await makeBodyParameter(stackArtifact, options.resolvedEnvironment, legacyAssets, options.toolkitInfo);
 
   await publishAssets(legacyAssets.toManifest(stackArtifact.assembly.directory), options.sdkProvider, stackEnv);
 
-  const changeSetName = options.changeSetName || 'cdk-deploy-change-set';
+  const shouldDisableRollback = (options.rollback === undefined && options.hotswap === true) || options.rollback === false;
+  options.rollback = shouldDisableRollback ? false : true;
```

Comment:
Not a fan of doing this here.  `tryHotswapDeployment ()` doesn't use this option, so dealing with it here is a little confusing to me. Let's move this down to `prepareAndExecuteChangeSet()`, where it's actually used.